File: src/main/java/ru/example/repository/BookRepository.java
package ru.example.repository;

import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;
import ru.example.model.Book;

@RepositoryRestResource(collectionResourceRel = "books", path = "books")
public interface BookRepository extends CrudRepository<Book, Long>, JpaSpecificationExecutor<Book> {
}

File: src/main/java/ru/library/repository/BookRepository.kt
package ru.library.repository

import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.repository.CrudRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import ru.library.model.Book

@RepositoryRestResource(collectionResourceRel = "books", path = "books")
interface BookRepository : CrudRepository<Book, Long>, JpaSpecificationExecutor<Book>